packages/playground/remote/service-worker.ts

```diff
@@ -27,8 +28,66 @@ if (!(self as any).document) {
 
 reportServiceWorkerMetrics(self);
 
+const cache = new WorkerCache(buildVersion);
+
+/**
+ * For offline mode to work we need to cache all required assets.
+ *
+ * These assets are listed in the `/assets-required-for-offline-mode.json` file
+ * and contain JavaScript, CSS, and other assets required to load the site without
+ * making any network requests.
+ */
+cache.cacheOfflineModeAssets();
+
+/**
+ * Cleanup old cache.
+ *
+ * We cache data based on `buildVersion` which is updated whenever Playground is built.
+ * So when a new version of Playground is deployed, the service worker will remove the old cache and cache the new assets.
+ *
+ * If your build version doesn't change while developing locally check `buildVersionPlugin` for more details on how it's generated.
+ */
+cache.cleanup();
```

Comment:
@adamziel I'm [exploring triggering caching on boot, but it's tricky](https://github.com/WordPress/wordpress-playground/pull/1623). Let's trigger caching in the worker for v1 and see how we can improve it later.

Reply:
> Maybe caching application assets for offline mode is best in the Service Worker. It is part of what Service Workers do.

I agree. We wanted to keep it in the SW, but wanted to trigger `cacheOfflineModeAssets` from boot.

Reply:
I agree, this is service worker specifiv and I don't think this is a wrong place to store the code. 

But triggering `cacheOfflineModeAssets` from boot has some benefits. It would allow boot to control what to cache.
For example, we only cache the current PHP and WP versions, if boot had control over it, we could add support for caching more versions.

Reply:
@bgrgicak I'd still keep it in the service worker. The service worker wouldn't do anything eagerly, it would only look for requests to `php.wasm` and `php.js` and then it would cache the responses. With that in place, the boot flow would only have to trigger a `fetch()`. In fact, it wouldn't be boot specific, we could trigger a few fetch() requests in the UI when the user clicks a "fetch all Playground assets" button.

Reply:
> With that in place, the boot flow would only have to trigger a fetch(). 

I will explore this in https://github.com/WordPress/wordpress-playground/pull/1623

Reply:
I have it working in https://github.com/WordPress/wordpress-playground/pull/1623.